### ISSUE TITLE
[bug/mechanics] Picking up minerals/cargo always says that 1 item = 1 ton

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#1450]
Was messing around with some custom minables, and many of the items that you can find have a mass other than 1. But, it will always say that I picked up `"<number of items> ton(s) of <item>."`
![screenshot_1](https://cloud.githubusercontent.com/assets/17688683/17651628/43410be2-6239-11e6-9d6f-89c5f02c9c54.png)
Despite the item actually weighing 8 tons a piece.
![screenshot_2](https://cloud.githubusercontent.com/assets/17688683/17651629/469c32da-6239-11e6-9a15-b1186f5715df.png)
It also lower cases the name. Kind of annoying in this case.

**Comment by KiLEdEnNis** [comment]
did you forget 3/4th of the battery pack when picking it up?
no wonder it doesn't work

**Comment by Amazinite** [comment]
I guess some of it got blasted off when the ship was destroyed. :)

**Comment by KiLEdEnNis** [comment]
it still looks pretty new though, sell it on eBay for practically new, easy 12000 credits

**Comment by Amazinite** [comment]
Yeah, I still need to do graphics for all the damaged outfits. xD

**Comment by Bladewood** [comment]
eBay still exists 10 centuries later?

**Comment by Sinsling** [comment]
eBay and Amazon, the only corporations to survive the turf wars. Sounds legit.

**Comment by jafdy** [comment]
But have they been swallowed up by the Syndicate or simply Overshadowed?

**Comment by KiLEdEnNis** [comment]
@jafdy I always knew it...

**Comment by Amazinite** [comment]
```
message = name + (amount == 1 ? "a ton" : Format::Number(amount) + " tons")
					+ " of " + Format::LowerCase(commodity) + ".";
```
Yeah, it assumes that everything you pick up is 1 ton, as amount is used to display the mass. I guess that's fair, given that that's how the vanilla game is.

**Comment by MarcelineVQ** [comment]
Good observation, this should probably be fixed to report fractional mass/space since it is currently inaccurate for ammo as well.

**Comment by Amazinite** [comment]
Yeah, I saw your issue and thought that they're probably related.

**Comment by EndrosG** [comment]
I strictly prefer the "units" way to go.
But I would, to make it a bit more fizzling, use "ton(s)" if UnitSize() == 1 and use "unit(s)" otherwise.
Your observation regarding ammo comes from line 1143ff.
Your damaged items are not installable anymore. I wonder why... You could easily allow for damaged stuff to be "installed". Then, you will always get the proper message from line 1149. That is, what happens with ammo.
And we dont need to tell the user in the first place that it were 15 tons of damaged LP036a... 2 units / pieces of damaged LP036a (345 tons of cargo space remaining) seems sufficient, imho.

That is, having a mission with a "landing" -> on accept "intact outfit" -1 && "damaged outfit" +1 would be funny on the one side, catastrophic on the other side if you get stranded on a location without outfitter. --- Can be compensated by some "repair" mission available in the spaceport then. LORE-wise: components of "outfit x" have been stolen. You better find the thefts... (well, this is not "damaged" anymore but "incomplete", but that's just a word...

**Comment by gunqqer** [comment]
So you're saying just the message to say "you pick up X damaged LP036a and have Y cargo space remaining"?

**Comment by EndrosG** [comment]
Yes and no.
I focused on the relevance of the "installable" attribute. If your "damaged" item is installable, it will automatically be shown as `you pick up X "damaged LP036a"`, but if it is `installable -1`, then it is treated as cargo and thus currently registered in tons. And for any cargo not having the mass of 1 ton I suggest using `unit(s)` instead of `ton(s)`.
If it is `installable` (default) or not is something also does influence, as far as I know, whether depreciation is applied or not. At least that's my current knowledge and thus I also added that in #1996 as part of my comment - depreciation affects balancing questions. Since #1996 is about minables in pretty exactly the same fashion as this issue, they are both referenced each other and I made a pretty long comment over there where I tried to bring quite some new input into designing this.
